tools/scripts/build_elf2uf2.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018, Niklas Hauser
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import fnmatch
+import shutil
+import subprocess
+import binascii
+import struct
+import sys
+import multiprocessing
+from pathlib import Path
+
+def repopath(path):
+	return (Path(os.path.abspath(__file__)).parents[2] / path)
+
+def run(where, command, stdin=None):
+	print(command)
+	result = subprocess.run(command, shell=True, cwd=where, input=stdin, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+	if result.returncode != 0:
+		print('failed exec command ' + command)
+		print(result.stderr.decode("utf-8").strip(" \n"))
+		exit(1)
+	return (result.returncode,
+	        result.stdout.decode("utf-8").strip(" \n"),
+	        result.stderr.decode("utf-8").strip(" \n"))
+
+def bitrev(x, width):
+    return int("{:0{w}b}".format(x, w=width)[::-1], 2)
+
+builddir = repopath('build/elf2uf2')
+if not os.path.exists(builddir):
+	os.mkdir(builddir)
+run(builddir,"cmake ../../ext/rp/pico-sdk/tools/elf2uf2 -G \"Unix Makefiles\"")
+run(builddir,'make') 
+
+bindir = repopath('build/bin')
+if not os.path.exists(bindir):
+	os.mkdir(bindir)
+
+shutil.copy(os.path.join(builddir,'elf2uf2'),bindir)
+
+exit(0)
```

Comment:
This seems very hacky? The `repopath("build/bin")` won't exist for external projects with a custom build path

Kinda sucks that they don't provide pre-compiled versions for Windows/Linux/macOS.
Then we could just copy the binaries and decide the architecture at runtime [like we do for CrashDebug](https://github.com/modm-io/modm/blob/develop/tools/build_script_generator/module.lb#L181-L184).

Is there a way to compile the binaries during the pico-sdk-partial update? Probably only for Linux though, but I can give you a manually compiled macOS binary (and we'll just ignore Windows for now) and then we can remove this step?

Reply:
I think I can just rewrite elf2uf2 in python